scripts/apps/authoring/authoring/directives/ItemCarouselDirective.ts

```diff
@@ -85,8 +85,10 @@ export function ItemCarouselDirective($timeout, notify) {
                     }
 
                     Promise.all([
-                        waitForImagesToLoad(carouselImages),
-                        waitForAudioAndVideoToLoadMetadata(carouselAudiosAndVideos),
+                        waitForMediaToLoad(carouselImages.filter(
+                            (img: HTMLImageElement) => img.complete === false), 'load'),
```

Comment:
Since you now have element-type specific behavior in `waitForMediaToLoad`, can you also move the logic of filtering the elements and determining the event type there as well?